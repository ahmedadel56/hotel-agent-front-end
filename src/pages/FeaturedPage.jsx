/* eslint-disable */
import React from 'react';
import { useSelector } from 'react-redux';
import FeaturedHotel from '../components/FeaturedHotel';
import Carousel from 'react-multi-carousel';
import 'react-multi-carousel/lib/styles.css';

const FeaturedPage = () => {
  const responsive = {
    desktop: {
      breakpoint: { max: 3000, min: 1024 },
      items: 3,
      slidesToSlide: 1, // optional, default to 1.
    },
    tablet: {
      breakpoint: { max: 1024, min: 464 },
      items: 2,
      slidesToSlide: 1, // optional, default to 1.
    },
    mobile: {
      breakpoint: { max: 464, min: 0 },
      items: 1,
      slidesToSlide: 1, // optional, default to 1.
    },
  };
  const { data } = useSelector((state) => state.MostRecent);

  return (
    <div className="featured-container">
      <div className="featured-heading">
        <h1>Featured Hotels</h1>
        <p>Please select a Hotel</p>
      </div>
      <Carousel
        swipeable={true}
        draggable={true}
        responsive={responsive}
        keyBoardControl={true}
        customTransition="all .5"
        containerClass="carousel-container"
        removeArrowOnDeviceType={['tablet', 'mobile']}
        itemClass="carousel-item-padding-40-px"
      >
        {data.map((hotel) => (
          <FeaturedHotel hotel={hotel} key={hotel.id} />
        ))}
      </Carousel>
    </div>
  );
};

export default FeaturedPage;
