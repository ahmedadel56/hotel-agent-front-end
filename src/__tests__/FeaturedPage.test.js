import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import FeaturedPage from '../pages/FeaturedPage';

it('Check if Featured page renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <FeaturedPage />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});

afterEach(() => {
  jest.clearAllMocks();
});
