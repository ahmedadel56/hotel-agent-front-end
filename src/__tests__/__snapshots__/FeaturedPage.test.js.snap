// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Featured page renders correctly 1`] = `
<div
  className="featured-container"
>
  <div
    className="featured-heading"
  >
    <h1>
      Featured Hotels
    </h1>
    <p>
      Please select a Hotel
    </p>
  </div>
  <div
    className="react-multi-carousel-list carousel-container "
    dir="ltr"
  >
    <ul
      className="react-multi-carousel-track "
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "overflow": "unset",
          "transform": "translate3d(0px,0,0)",
          "transition": "none",
        }
      }
    />
  </div>
</div>
`;
